plugin.audio.podcasts/addon.py

```diff
@@ -0,0 +1,327 @@
+
+            if settings.getSetting("group_%i_enable" % g) == "false":
+                continue
+
+            entries = []
+            for e in range(self._ENTRIES):
+
+                if settings.getSetting("group_%i_rss_%i_enable" % (g, e)) == "false":
+                    continue
+
+                icon = settings.getSetting("group_%i_rss_%i_icon"
+                                                % (g, e))
+
+                entries += [{
+                    "path": "pod_%i" % e,
+                    "name": settings.getSetting("group_%i_rss_%i_name"
+                                                % (g, e)),
+                    "icon": icon if icon != "" else "icon.png",
+                    "params": [
+                        {
+                            "call": "renderRss",
+                            "url": settings.getSetting("group_%i_rss_%i_url"
+                                                       % (g, e))
+                        }
+                    ],
+                    "node": []
+                }]
+
+            groups += [{
+                "path": "group_%i" % g,
+                "name": settings.getSetting("group_%i_name" % g),
+                "node": entries
+            }]
+
+        self._menu = [
+            {  # root
+                "path": "",
+                "node": groups
+            }
+        ]
+
+    def _load_opml(self, path):
+
+        try:
+            with open(path) as _opml_file:
+                _data = _opml_file.read()
+                return xmltodict.parse(_data)
+        except:
+            xbmc.log("Cannot open opml file", xbmc.LOGERROR)
+            return None
+
+    def _loadRss(self, url):
+
+        res = requests.get(url)
+        if res.status_code == 200 and res.text.startswith("<?xml"):
+            return xmltodict.parse(res.text)
+
+        elif res.status_code != 200:
+            raise HttpStatusError(
+                "Unexpected HTTP Status %i for podcast %s" % (res.status_code, url))
+
+        else:
+            raise HttpStatusError("Unexpected content for podcast %s" % url)
+
+    def _load_rss_feed(self, url):
+
+        rss_feed = self._loadRss(url)
+
+        channel = rss_feed["rss"]["channel"]
+
+        if "image" in channel and "url" in channel["image"]:
+            image = channel["image"]["url"]
+        elif "itunes:image" in channel:
+            image = channel["itunes:image"]["@href"]
+        else:
+            image = ""
+
+        items = channel["item"]
+
+        return items, image
+
+    def playRss(self, parent, path, params):
+
+        try:
+            url = params["url"][0]
+            items, image = self._load_rss_feed(url)
+            index = int(params["index"][0])
+
+            item = items[index]
+            if "enclosure" in item:
+                stream_url = item["enclosure"]["@url"]
+            else:
+                stream_url = item["guid"]
+
+            xbmc.executebuiltin('PlayMedia(%s)' % stream_url)
```

Comment:
I checked and a plugin should use xbmcplugin.setResolvedUrl() in combination with xbmcgui.ListItem.setProperty('IsPlayable', 'true'). That way Kodi will also remember what items were played in the past. But I guess this works. Same in other places.

Reply:
Could you change this to use `xbmcplugin.setResolvedUrl()`